internal/clients/stubs/gcp_stub.go

```diff
@@ -60,19 +79,23 @@ func (mock *GCPClientStub) Status(ctx context.Context) error {
 }
 
 func (mock *GCPClientStub) GetInstanceDescriptionByID(ctx context.Context, id, zone string) (*clients.InstanceDescription, error) {
```

Comment:
I'd expect this method to actually search if the Instance is in the `Instaces` list, so it could actually fail if the instance does not exists.... creating the instance in `Get` method sounds weird :thinking: